js/cart.js

```diff
@@ -19,25 +19,58 @@ function renderCart() {
 }
 
 // TODO: Remove all of the rows (tr) in the cart table (tbody)
-function clearCart() {}
+function clearCart() {
+  var table = document.getElementById('cart');
+
+  var tbody = table.childNodes;
+  console.log(tbody[0].childNodes);
+  //var nodes = tbody.firstChild;
+
+  
+
+  for (var i = 0; i < cart.items.length; i++) {
+    //tbody.removeChild(tbody.firstChild);
+  }
+}
 
 // TODO: Fill in the <tr>'s under the <tbody> for each item in the cart
 function showCart() {
-
   // TODO: Find the table body
+  var tbody = document.getElementsByTagName('tbody')[0];
+
 
-  // TODO: Iterate over the items in the cart
-  // TODO: Create a TR
-  // TODO: Create a TD for the delete link, quantity,  and the item
   // TODO: Add the TR to the TBODY and each of the TD's to the TR
+  // TODO: Iterate over the items in the cart
+  for (var i = 0; i < cart.items.length; i++) {
+    // TODO: Create a TR
+    var tr = document.createElement('tr');
+    tr.setAttribute('id', i);
+    tbody.appendChild(tr);
+
+    // TODO: Create a TD for the delete link, quantity,  and the item
+    var deleteTd = document.createElement('td');
+    deleteTd.textContent = 'X';
+    tr.appendChild(deleteTd);
 
+    var quantityTd = document.createElement('td');
+    quantityTd.textContent = cart.items[i].quantity;
+    tr.appendChild(quantityTd);
+
+    var itemTd = document.createElement('td');
+    itemTd.textContent = cart.items[i].product;
+    tr.appendChild(itemTd);
+  }
 }
 
 function removeItemFromCart(event) {
-
   // TODO: When a delete link is clicked, use cart.removeItem to remove the correct item
+  if (event.target.textContent === 'X') {
```

Comment:
This is fine for the assignment, but try to imagine a way of using CSS selectors to determine if the appropriate tag has been clicked. That textContent value may change over time and this statement really ties down the Cart app to the magic 'X'. Whereas a structural change in HTML will often NOT happen until side-effects are fully investigated.